services/agents/registry.go

```diff
@@ -312,23 +314,39 @@ func (r *Registry) handleJobResult(l *logrus.Entry, result *agentpb.JobResult) {
 				return errors.Errorf("result type %s doesn't match job type %s", models.MySQLBackupJob, res.Type)
 			}
 
-			_, err := models.UpdateArtifact(t.Querier, res.Result.MySQLBackup.ArtifactID, models.UpdateArtifactParams{
+			artifact, err := models.UpdateArtifact(t.Querier, res.Result.MySQLBackup.ArtifactID, models.UpdateArtifactParams{
 				Status: models.BackupStatusPointer(models.SuccessBackupStatus),
 			})
 			if err != nil {
 				return err
 			}
+
+			if artifact.Type == models.ScheduledArtifactType {
+				go func() {
+					if err := r.retentionService.EnforceRetention(ctx, artifact.ScheduleID); err != nil {
```

Comment:
This call has no querier argument, so in my opinion it shouldn't be inside of the transaction. Because in case of transaction rollback this call shouldn't happen.

Reply:
any suggestion how to properly do it? new switch outside transaction?
But then I'll need again to select artifact.

Reply:
Or maybe pass querier to it and remove DB from retention service?